src/services/TestResultsService.ts

```diff
@@ -343,34 +348,46 @@ export class TestResultsService {
     const adrTestTypeIds = ["41", "64", "65", "66", "67", "95", "102", "103", "104"];
     return adrTestTypeIds.includes(testType.testTypeId);
   }
-  public getMostRecentExpiryDateOnAllTestTypesByVin(vin: any) {
-    let maxDate = new Date(1970, 1, 1);
+  public getMostRecentExpiryDateOnAllTestTypesByVin(vin: string): Promise<IMaxDates> {
+    const maxDate = new Date(1970, 1, 1);
+    const maxDatesForTestTypes: IMaxDates = {maxDateForAllTestTypes: maxDate, maxDateForAnnualTestTypes: maxDate};
+
     return this.getTestResults({
       vin,
       testStatus: TEST_STATUS.SUBMITTED,
       fromDateTime: new Date(1970, 1, 1),
       toDateTime: new Date()
     })
         .then((testResults) => {
-          const filterTestTypes: any[] = [];
+          const filteredTestTypes: {allTestTypes: any[]; annualTestTypes: any[]} = {allTestTypes: [], annualTestTypes: []};
           testResults.forEach((testResult: { testTypes: any; vehicleType: any; vehicleSize: any; vehicleConfiguration: any; noOfAxles: any; }) => {
-            testResult.testTypes.forEach((testType: { testExpiryDate: string; }) => {
-              if (testType.testExpiryDate) {
-                filterTestTypes.push(testType);
+            testResult.testTypes.forEach((testType: { testExpiryDate: string; testCode: string; }) => {
+              // prepare a list of annualTestTypes and allTestTypes with expiry.
+              if (TestResultsService.isValidTestCodeForExpiryCalculation(testType.testCode.toUpperCase()) && testType.testExpiryDate) {
+                filteredTestTypes.annualTestTypes.push(testType);
```

Comment:
a bit confused by this piece of logic. Instinct suggests this should be testing for "isAnnualTestCode..." if it's going on the "annual test types" list. Also, everything should go on the "all test types" list, so no "else" required.
Probably should also split this out into a named function so it's clear WHY the splitting is happening.